sys/ztimer/core.c

```diff
@@ -406,3 +406,31 @@ static void _ztimer_print(const ztimer_clock_t *clock)
     } while ((entry = entry->next));
     puts("");
 }
+
+uint32_t ztimer_until(ztimer_clock_t *clock, ztimer_t *timer)
+{
+    unsigned state = irq_disable();
+
+    if (!_is_set(clock, timer)) {
+        irq_restore(state);
+        return 0;
+    }
+    uint32_t sum = clock->list.offset;
```

Comment:
Would this be easier if `_ztimer_update_head_offset` would be called? Unless we're just in an interrupt-free state with pending timers, that'd just update the head pointer and first delta -- and then it is known that the clock base is "now", and summation would be just over what's left in the timer queue.

Reply:
It might, I'd prefer though if this wouldn't change the clock's state.

Reply:
i agree with kasper030 and like to avoid changing the list (have this a reader not a writer)

Reply:
OK.

Might still have some potential for simplification, but that's on other lines.

Reply:
thanks for spotting the error that was caused by not doing the head update